woocommerce/woocommerce
# Display a notice for expired/expiring subscriptions along with a plugin update message.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces the Feature of Showing an expiry subscription message in the plugin table list for WooCommerce extensions that have either expired or are expiring.



### If you haven't configured your development environment for WooCommerce development, please refer to this documentation.
- https://github.com/woocommerce/woocommerce/blob/trunk/README.md#getting-started
- https://github.com/woocommerce/woocommerce/blob/trunk/DEVELOPMENT.md#plugin-development-environments




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build WooCommerce on your local environment, e.g.: `cd plugins/woocommerce` then `nvm use && pnpm run build --filter=woocommerce`.
2. Connected your store with local woocommerce.com test site. 
3. Download an old version of a Woo Plugins (for example: Checkout Field Editor version 1.7.10). You can find it in the github/woocommerce account or My account > Download page from local woocom test site.
4. Make sure you have active subscription for this plugin.
5. Upload and install the old plugin
6. Make sure you have no expired subscriptions and no subscriptions expiring in the next 7 days
7. Go to the plugin list page and confirm you see no messsage related to subscriptions.
8. Set one of the orders as expiring soon by setting an expiry date in the next 7 days. To do that, run the [following query](https://github.com/Automattic/woocommerce.com/issues/20256#issuecomment-2111773279) and make sure Auto Renew is disabled for this subscription


9. Visit the Woocommerce > Extensions > My subscription page and click the `refresh` button 
10. Now visit the plugin list page again, and should see the notice about the expiring subscription 

![Screenshot 2024-05-01 at 1 12 51 PM](https://github.com/woocommerce/woocommerce/assets/10769293/19623e15-5e5a-403b-bca8-8b5d97d0958d)

11. For expired notice use the following query 

12. Follow the steps 9 and 10 again, and should see this message.

![Screenshot 2024-05-01 at 1 25 30 PM](https://github.com/woocommerce/woocommerce/assets/10769293/833fd251-d5d1-4996-97ea-cec352e4db34)






### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
